.github/workflows/build-test-distribute.yaml

```diff
@@ -24,20 +28,21 @@ jobs:
         with:
           path: |
             ${{ env.CI_TOOLS_DIR }}
-          key: ${{ runner.os }}-devtools-${{ hashFiles('mk/dependencies/deps.lock') }}
+          key: devtools-${{ env.RUN_FULL_MATRIX && 'full-matrix' || 'normal' }}-${{ runner.os }}-${{ hashFiles('mk/dependencies/deps.lock') }}
           restore-keys: |
-            ${{ runner.os }}-devtools
+            devtools-${{ runner.os }}-${{ env.RUN_FULL_MATRIX && 'full-matrix' || 'normal' }}
+            devtools-${{ runner.os }}
       - run: |
           make dev/tools
-      - run: |
-          make clean
```

Comment:
Still need this to do a proper gen check

Reply:
Got it, I'll revert this change. 
I saw a failure caused by missing a zz_generated file in KM repo. I fixed it by removing `clean`. Now I think the correct fix should be fixing the installation of `controller-gen` tool so that it installs the tool correctly and generate the files as expected.